mitosheet/src/components/taskpanes/Graph/GraphSidebar.tsx

```diff
@@ -119,86 +120,73 @@ const GraphSidebar = (props: {
     dfNames: string[];
     graphSidebarSheet: number;
-    columnDtypesMap: Record<string, string>;
+    columnDtypesMap: Record<string, string> | undefined;
     mitoAPI: MitoAPI;
     setUIState: React.Dispatch<React.SetStateAction<UIState>>;
-    model_id: string;
-    lastGraphParams: Record<number, GraphParams | undefined>;
-    setLastGraphParams: (sheetIndex: number, graphParams: GraphParams) => void;
+    graphDataJSON: GraphDataJSON
+    lastStepIndex: number
 }): JSX.Element => {
 
-    const [selectedSheetIndex, _setSelectedSheetIndex] = useState(props.graphSidebarSheet);
-
-    // A wrapper around changing the selected sheet index that makes sure
-    // the correct sheet is displayed, and also loads the most recent graph
-    // from this sheet if it exists
-    const setSelectedSheetIndex = (newSelectedSheetIndex: number): void => {
-        const newParams = getGraphParams(props.lastGraphParams, newSelectedSheetIndex, props.sheetDataArray);
-        // Note we update the sheet before the graph parameters
-        _setSelectedSheetIndex(newSelectedSheetIndex);
-        _setGraphParams(newParams)
-        props.setUIState(prevUIState => {
-            return {
-                ...prevUIState,
-                selectedSheetIndex: newSelectedSheetIndex
-            }
-        });
-    }
-
-    // When opening the graphing modal, if there are params for the last graph that was made
-    // for this sheet, then take them. Otherwise, just take the default params
-    const [graphParams, _setGraphParams] = useState<GraphParams>(
-        getGraphParams(props.lastGraphParams, selectedSheetIndex, props.sheetDataArray)
-    );
-
-    // When we update the graph params, we also update the lastGraphParams in the 
-    // main Mito component, so that we can open the graph to the same state next time
-    const setGraphParams = (newGraphParams: GraphParams): void => {
-        _setGraphParams(newGraphParams);
-        props.setLastGraphParams(selectedSheetIndex, newGraphParams);
-    }
+    // We keep track of the graph data separately from the backend state so that 
+    // the UI updates imidietly, even though the backend takes a while to process.
+    const [graphParams, setGraphParams] = useState<GraphParams>(() => getGraphParams(props.graphDataJSON, props.graphSidebarSheet, props.sheetDataArray))
+    const dataSourceSheetIndex = graphParams.graphCreation.sheet_index
+    const graphScript = props.graphDataJSON[dataSourceSheetIndex]?.graphScript
+    const graphHTML = props.graphDataJSON[dataSourceSheetIndex]?.graphHTML
+    const [_copyGraphCode, graphCodeCopied] = useCopyToClipboard(props.graphDataJSON[dataSourceSheetIndex]?.graphGeneratedCode || '');
+    const [stepID, setStepID] = useState<string|undefined>(undefined);
 
-    const [graphObj, setGraphObj] = useState<GraphObject | undefined>(undefined);
     const [loading, setLoading] = useState<boolean>(false)
+    const [changeLoadingGraph] = useDelayedAction(LOAD_GRAPH_TIMEOUT)
```

Comment:
I think we should move this logic to using the `useDebouncedEffect` - which we added after this logic. Thoughts?